### ISSUE TITLE
Support DAP's `terminate` request when ending a debug session

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#54384]
To fix issue https://github.com/Microsoft/vscode-node-debug/issues/1 we will add this feature to VS Code:

- if a debug adapter supports "terminate", VS Code's red terminate button will first send a "terminate" request to the DA (which some DAs might implement by sending a `SIGINT` signal to the debuggee). VS Code remembers in its debug session state that a "terminate" request has been issued.
- If the debuggee has no `SIGINT` handler, the debuggee will terminate and end the debug session. This is almost identical to the current behaviour.
- if the debuggee has a `SIGINT` handler but doesn't terminate properly, the debug session will continue. Pressing the red terminate button again will send a "disconnect" request. This will kill the debuggee forcefully (like the current behaviour).

@isidorn I've added a terminate to VA Code's DAP.

**Comment by isidorn** [comment]
I have pushed the inital changes needed for this.
We still only have one action "Debug: Stop" which when invoked only once sends the terminate request (if the capability is supported). Otherwise if the capability is not supported, or if triggered for the second time sends a "disconnect".

Things to keep an eye open for: sessions staying around when debugging is over. Also extensinos host debugging, restarting and reloading. I tested this out but we should test more.
